File: Language Learning/Java/EclipseWorkspace/Lista15Ricardo/src/Exercicio_3_15/ContaBancaria.java
package Exercicio_3_15;

public class ContaBancaria {
	private double saldo;
	private String titular;
	
	public ContaBancaria() {
		this.saldo = 0;
		this.titular = "Fulano";
	}
	public ContaBancaria(double saldo, String titular) {
		this.saldo = saldo;
		this.titular = titular;
	}
	public void Depositar(double valor) {
		this.saldo += valor;
	}
	public double Sacar(double valor) {
		try {
			if(valor<=this.saldo) {
			this.saldo -= valor;
			return valor;
			}else {
				throw new SaldoInsuficiente(this.saldo, valor);
			}
			
		} catch (Exception e) {
			return 0;
		}
	}
	public String getTitular() {
		return this.titular;
	}
	public double getSaldo() {
		return this.saldo;
	}
}

File: Language Learning/Random Exercises/Prof Ricardo/Lista 15 Python/Exercicio_3_15/ContaBancaria.py
class SaldoInsuficiente:
    def __init__(self) -> None:
        self.err = "vc nao tem dinheiro"
    def __str__(self) -> str:
        return self.err
    
class ContaBancaria:
    
    def __init__ (self, nome):
        self.__nome = nome
        self.__saldo = 0
        self.__err = SaldoInsuficiente()
    def __str__(self):
        return "\n\nNome: " + self.__nome + "\nSaldo: " + str(self.__saldo)

    def depositar(self, quantidade):
        self.__saldo += quantidade
    def sacar(self, quantidade):
        if quantidade>self.__saldo:
            print(self.__err)
            return 0
        else:
            self.__saldo -= quantidade
            return quantidade
            
conta = ContaBancaria("Pedro")
print(conta)
conta.sacar(10)